README.md

```diff
@@ -144,6 +161,104 @@ module.exports = ({ renderJsBundles, renderCssLinks, production }) => {
 };
 ```
 
+## Deployment environments
+
+### Env vars
+
+Underreact allows you to inject environment variables **during the build time**. You can set them up by creating a `.env` file at the root of your project:
+
+```
+#.env
+SERVER=https://example.com
+```
+
+You can then use them in your codebase:
+
+```javascript
+const url = process.env.SERVER;
+fetch(`${url}/data`);
+```
+
+Any env variable that you use in your app must be declared in the `.env` file. This is required to prevent unintentional leaking of env vars in your javascript bundle. 
+
+If you want to pass an environment variable in the CLI, make sure it is declared in your `.env` file first and use it like:
+
+```
+SERVER=https://foobar.com npx underreact start
+```
+
+You can also expand already existing env variables in your machine (using [dotenv-expand](https://github.com/motdotla/dotenv-expand))
+
+```
+APP_VERSION=${npm_package_version}
+```
+
+Or expand local variables
+
+```
+SERVER=https://example.com
+SERVER_FOO=$SERVER/foo
+```
+
+### Targeting multiple deployment environments
+
+Underreact allows your app to target different environments with the help of the env var `DEPLOY_ENV`. There are certain rules on how `DEPLOY_ENV` behaves:
+
+- If `DEPLOY_ENV` is not set it will default to `development` in development mode and `production` in production mode.
+- You can set `DEPLOY_ENV` to any value you wish to better align with your target environments.
+- `DEPLOY_ENV` is not the same as `NODE_ENV`, refer to [Why not use NODE_ENV](#why-not-use-node_env).
+
+Underreact allows you to have multiple `.env` files for different deployment targets. It can read the following type of `.env` files
+
+- `.env:` Default.
+- `.env.development`, `.env.staging`, `.env.production`: Deployment-specific settings.
+
+The deployment target is defined by the env variable `DEPLOY_ENV`. Underreact would then find the file which matches the `.env.<DEPLOY_ENV>`. For example if your machine has `DEPLOY_ENV=staging`, underreact would try to find `.env.staging`. 
+
+**Underreact would always first load `.env` file** before loading the deployment specific `.env.<DEPLOY_ENV>`. This allows for overriding env variables mentioned in `.env` with the ones in `.env.<DEPLOY_ENV>`. The following example illustrates the overriding of env variables:
```

Comment:
Instead of explaining this as "first" and "second" I think we could cut right to the point and say that values in `.env.<DEPLOY_ENV>` will override values in `.env.`

It's also worth pointing out, while explaining overrides, that the `.env.<DEPLOY_ENV>` files do not need to contain *all* the variables in `.env`, only the ones that are being overridden for that DEPLOY_ENV.